presto-oracle/src/test/java/com/facebook/presto/plugin/oracle/TestOracleTypes.java

```diff
@@ -0,0 +1,142 @@
+        oracle.start();
+    }
+    public TestOracleTypes()
+    {
+        this(new TestingOracleServer());
+    }
+    private TestOracleTypes(TestingOracleServer oracleServer)
+    {
+        super(() -> createOracleQueryRunner(oracleServer));
+        this.oracleServer = oracleServer;
+    }
+
+    @AfterClass(alwaysRun = true)
+    public final void destroy()
+    {
+        if (oracleServer != null) {
+            oracleServer.close();
+        }
+    }
+
+    private DataSetup prestoCreateAsSelect(String tableNamePrefix)
+    {
+        return new CreateAsSelectDataSetup(new PrestoSqlExecutor(getQueryRunner()), tableNamePrefix);
+    }
+
+    @Test
+    public void testBooleanType()
+    {
+        DataTypeTest.create()
+                .addRoundTrip(booleanOracleType(), true)
+                .addRoundTrip(booleanOracleType(), false)
+                .execute(getQueryRunner(), prestoCreateAsSelect("boolean_types"));
+    }
+
+    @Test
+    public void testSpecialNumberFormats()
+    {
+        oracleServer.execute("CREATE TABLE test (num1 number)");
+        oracleServer.execute("INSERT INTO test VALUES (12345678901234567890.12345678901234567890123456789012345678)");
+        assertQuery("SELECT * FROM test", "VALUES (12345678901234567890.1234567890)");
+    }
+
+    @Test
+    public void testVarcharType()
+    {
+        DataTypeTest.create()
+                .addRoundTrip(varcharDataType(10), "test")
+                .addRoundTrip(stringDataType("varchar", createVarcharType(4000)), "test")
+                .addRoundTrip(stringDataType("varchar(5000)", createUnboundedVarcharType()), "test")
+                .addRoundTrip(varcharDataType(3), String.valueOf('\u2603'))
+                .execute(getQueryRunner(), prestoCreateAsSelect("varchar_types"));
+    }
+
+    private static DataType<Boolean> booleanOracleType()
+    {
+        return DataType.dataType(
+                "boolean",
+                BigintType.BIGINT,
+                val -> val ? "1" : "0",
+                val -> val ? 1L : 0L);
+    }
+
+    private static DataType<BigDecimal> numberOracleType(DecimalType type)
+    {
+        String databaseType = format("decimal(%s, %s)", type.getPrecision(), type.getScale());
+        return numberOracleType(databaseType, type);
+    }
+
+    private static <T> DataType<T> numberOracleType(String inputType, Type resultType)
+    {
+        Function<T, ?> queryResult = (Function<T, Object>) value ->
+                (value instanceof BigDecimal && resultType instanceof DecimalType)
+                        ? ((BigDecimal) value).setScale(((DecimalType) resultType).getScale(), HALF_UP)
+                        : value;
+
+        return DataType.dataType(
+                inputType,
+                resultType,
+                value -> format("CAST('%s' AS %s)", value, resultType),
+                queryResult);
+    }
+
+    public static DataType<LocalDate> dateOracleType()
```

Comment:
many functions in this class are not used; just remove

Reply:
removed